File: app/src/main/java/com/rendersoncs/reportform/util/RVEmptyObserver.java
package com.rendersoncs.reportform.util;

import androidx.recyclerview.widget.RecyclerView;
import android.view.View;

public class RVEmptyObserver extends RecyclerView.AdapterDataObserver {
    public View emptyView;
    public View floatingActionButton;
    private RecyclerView recyclerView;

    public RVEmptyObserver(RecyclerView rv, View ev, View fb){
        this.recyclerView = rv;
        this.emptyView = ev;
        this.floatingActionButton = fb;
        checkEmpty();
    }

    private void checkEmpty() {
        if (emptyView != null && recyclerView.getAdapter() != null){
            boolean emptyViewVisible = recyclerView.getAdapter().getItemCount() == 0;
            emptyView.setVisibility(emptyViewVisible ? View.VISIBLE : View.GONE);
            recyclerView.setVisibility(emptyViewVisible ? View.GONE : View.VISIBLE);
        }

        if (floatingActionButton != null && recyclerView.getAdapter() != null){
            boolean emptyViewVisible5 = recyclerView.getAdapter().getItemCount() == 0;
            floatingActionButton.setVisibility(emptyViewVisible5 ? View.GONE : View.VISIBLE);
            recyclerView.setVisibility(emptyViewVisible5 ? View.GONE : View.VISIBLE);
        }
    }

    public void onChanged(){ checkEmpty(); }
    public void onItemRangeInserted(int positionStart, int itemCount) { checkEmpty(); }
    public void onItemRangeRemoved(int positionStart, int itemCount) { checkEmpty(); }
}

File: app/src/main/java/com/rendersoncs/reportform/view/services/util/RVEmptyObserver.kt
package com.rendersoncs.reportform.view.services.util

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.AdapterDataObserver

class RVEmptyObserver(private val recyclerView: RecyclerView,
                      private val emptyView: View?,
                      private val floatingActionButton: View?) :
        AdapterDataObserver() {

    private fun checkEmpty() {
        if (emptyView != null && recyclerView.adapter != null) {
            val emptyViewVisible = recyclerView.adapter!!.itemCount == 0
            emptyView.visibility = if (emptyViewVisible) View.VISIBLE else View.GONE
            recyclerView.visibility = if (emptyViewVisible) View.GONE else View.VISIBLE
        }
        if (floatingActionButton != null && recyclerView.adapter != null) {
            val emptyViewVisible5 = recyclerView.adapter!!.itemCount == 0
            floatingActionButton.visibility = if (emptyViewVisible5) View.GONE else View.VISIBLE
            recyclerView.visibility = if (emptyViewVisible5) View.GONE else View.VISIBLE
        }
    }

    override fun onChanged() {
        checkEmpty()
    }

    override fun onItemRangeInserted(positionStart: Int, itemCount: Int) {
        checkEmpty()
    }

    override fun onItemRangeRemoved(positionStart: Int, itemCount: Int) {
        checkEmpty()
    }

    init {
        checkEmpty()
    }
}